### PR TITLE
Replace `docker-compose` v1 command with `docker compose` v2

### PR DESCRIPTION
**Pull request by maxproske**
Replace legacy `docker-compose` command with `docker compose`  (no hyphen) in all examples https://github.com/vercel/next.js/pull/36283#pullrequestreview-946487627

## Documentation / Examples

- [X] Make sure the linting passes by running `pnpm lint`
- [X] The examples guidelines are followed from [our contributing doc](https://github.com/vercel/next.js/blob/canary/contributing.md#adding-examples)

### CODE REVIEW

File: examples/with-docker-compose/.env [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # DO NOT ADD SECRETS TO THIS FILE. This is a good place for defaults.
-# If you want to add secrets use `.env.production.local` instead, which is automatically detected by docker-compose.
+# If you want to add secrets use `.env.production.local` instead, which is automatically detected by docker compose.
```
Comment (balazsorban44) [inline examples/with-docker-compose/.env:2]:
```suggestion
# If you want to add secrets use `.env.production.local` instead, which is automatically detected by `docker compose`.
```

**Comment by balazsorban44** [comment]
I wonder in which version of Docker this was added. I'm on `Docker version 20.10.17, build 100c701` and `docker compose` yields:

> docker: 'compose' is not a docker command.
See 'docker --help'

https://docs.docker.com/engine/release-notes/#201017 This is a ~month old release

**Comment by Yhozen** [comment]
@balazsorban44 the beta is [more than a year old](https://www.docker.com/blog/volume-management-compose-v2-skipping-updates-and-more-in-docker-desktop-3-4/) and compose is a [docker cli plugin](https://docs.docker.com/compose/install/compose-plugin/) now. I have been using it for more than 4 months now without any problem

**Comment by balazsorban44** [comment]
hmm, so it's *not* part of `docker` by default but a plugin you have to install in addition. should we mention that since people might run into this like me? thus wasn't clear for me.

**Comment by Yhozen** [comment]
@balazsorban44 it installed [by default on Docker Desktop version but not on the Linux cli version](https://www.docker.com/blog/announcing-compose-v2-general-availability/#:~:text=What%20do%20I%20need%20to%20do%20as%20a%20developer?) 🤷🏻‍♂️ weird decisions

**Comment by maxproske** [comment]
Closing until Docker allows us to alias `docker compose` to `docker-compose`

**Pull request closed without merging**